doc/docs/api.md

```diff
@@ -714,23 +762,52 @@ tdb_error tdb_event_filter_get_item(const struct tdb_event_filter *filter,
-* `item_index` item index in the clause.
+* `clause_index` clause index: `0 <= clause_index < tdb_event_filter_num_clauses()`.
+* `item_index` item index in the clause: `0 <= term_index < tdb_event_filter_num_terms()`.
 * `item` returned item.
 * `is_negative` return 1 if the item negative, 0 otherwise, as set in [tdb_event_filter_add_term](#tdb_event_filter_add_term).
 
-Returns 0 if an item was found at this location, `TDB_ERR_NO_SUCH_ITEM` otherwise. Note
-that empty clauses always return `TDB_ERR_NO_SUCH_ITEM` although the clauses themselves
-are valid.
+Returns 0 if an item was found at this location and is a match term. If the clause or term
+do no exist, `TDB_ERR_NO_SUCH_ITEM` is returned.  Note that empty clauses always return
+`TDB_ERR_NO_SUCH_ITEM` although the clauses themselves are valid. Lastly, if you try to call
+`tdb_event_filter_get_item` on a time-range term, then `TDB_ERR_INCORRECT_TERM_TYPE` is
+returned.
+
+### tdb_event_filter_get_time_range
+Get a time-range term from a clause in this filter.
+```c
+tdb_error tdb_event_filter_get_time_range(const struct tdb_event_filter *filter,
+                                          uint64_t clause_index,
+                                          uint64_t term_index,
+                                          uint64_t *start_time,
+                                          uint64_t *end_time)
+```
+* `filter` filter handle.
+* `clause_index` clause index: `0 <= clause_index < tdb_event_filter_num_clauses()`.
+* `term_index` term index in the clause: `0 <= term_index < tdb_event_filter_num_terms()`.
+* `start_time` start time (inclusive) of the time range.
+* `end_time` end_time (exclusive) of the time range.
+
+Returns 0 if a time-range term was found at this location. If the clause or term do no exist,
+`TDB_ERR_NO_SUCH_ITEM` is returned.  Note that empty clauses always return
+`TDB_ERR_NO_SUCH_ITEM` although the clauses themselves are valid. Lastly, if you try to call
+`tdb_event_filter_get_time_range` on a match term, then `TDB_ERR_INCORRECT_TERM_TYPE` is
+returned.
+
 
 Here is an example how to deconstruct a filter back to clauses and items:
 ```c
 for (clause = 0; clause < tdb_event_filter_num_clauses(filter); clause++){
-    uint64_t item, idx = 0;
+    uint64_t item, start_time, end_time, idx = 0;
+    tdb_event_filter_term_type term_type;
     int is_negative;
-    while (!tdb_event_filter_get_item(filter, clause, idx, &item, &is_negative)){
-        /* do something with 'item' in 'clause' */
-        ++idx;
+    for (term = 0; term < tdb_event_filter_num_terms(filter, clause); term++){
+        tdb_event_filter_get_term_type(filter, clause, term, term_type);
+        if (type == TDB_EVENT_FILTER_MATCH_TERM){
+            /* do something with 'item' at 'term' in 'clause' */
+        } else if(type == TDB_EVENT_FILTER_TIME_RANGE_TERM){
+            /* do something with 'start_time' and 'end_time' at 'term' in 'clause' */
```

Comment:
add a call to `tdb_event_filter_get_time_range` here and `tdb_event_filter_get_item` above